base/reduce.jl

```diff
@@ -131,7 +131,13 @@ mapfoldr(f, op, v0, itr) = mapfoldr_impl(f, op, v0, itr, endof(itr))
 Like `mapfoldr(f, op, v0, itr)`, but using the first element of `itr` as `v0`. In general,
 this cannot be used with empty collections (see `reduce(op, itr)`).
 """
-mapfoldr(f, op, itr) = (i = endof(itr); mapfoldr_impl(f, op, f(itr[i]), itr, i-1))
+function mapfoldr(f, op, itr)
+    i = endof(itr)
+    if i == 0
```

Comment:
I copied the check from `mapfoldr_impl` ([here](https://github.com/vsart/julia/blob/222810b7fdc868949b4b6d364f0725df9133f5c8/base/reduce.jl#L107)) which tests if the `i` argument is 0, I'm assuming the one I implemented in `mapfoldr` should change so should that check?